src/modules/core/database/prisma/prisma.service.ts

```diff
@@ -111,4 +119,26 @@ export class PrismaService
     }
     return;
   }
+
+  /**
+   * 매핑테이블의 매핑된 데이터 정보 유효성 검사를 해주는 메서드
+   */
+  // 매핑 테이블 과제 id, 해시태그 id
+  async validateMappedData(
+    modelName: PrismaModelName,
+    where: Record<string, number | { in: number[] }>,
+    length: number,
```

Comment:
```suggestion
  async validateMappedData<T extends LessonHashtagMapping | Major>(
    modelName: PrismaModelName,
    where: {
      [key in keyof Omit<T, 'id' | 'createdAt'>]: number | { in: number[] };
    },
    length: number,
```
1. 지금 구현된 메서드가 범용적으로 쓸수있는 대신 타입 자체가 너무 약하다는 생각이 들어요 개인적으로 번거롭더라도 이런식으로라도 되면 어떨까 하는데 고려해주시면 감사하겠습니다.
2. 3번째 인자로 받는 length 가 사실상 number[] 의 length 와 같은데 굳이 받아야하는지 의문이 있네요.
3. 구현체를 보면 조회 후 예상하는 값들이 모두 있는지 확인하는거같은데 메서드명만 가지고는 아래 구현체의 역할을 표현하기에는 부족하다는 생각이 듭니다.

Reply:
리뷰 해주신 타입으로 타입을 구체화 시키는 것에 대해서 공감하며 동의합니다.
3번째 인자로 받는 length가 where 조건에 두번째 인자로 받는 number[]의 length 와 같지만 두번째 인자로 받는 인자의 이름이 런타임에 결정되기 때문에 따로 length를 받아야 했습니다 ㅠㅠ